source/Ship.h

```diff
@@ -167,6 +167,9 @@ class Ship : public Body, public std::enable_shared_from_this<Ship> {
 	// This is only useful for the player's ships.
 	double Attraction() const;
 	double Deterrence() const;
+	// Get and set if this ship will take the initiative to fire at enemies.
+	void SetHoldFire(bool hold = true);
+	bool HoldingFire() const;
```

Comment:
Not a fan of storing hold fire information on the Ship, especially as it requires that const cast to actually change the value. AI already contains `std::map<const Ship *, Orders> orders;`. Could you not check if the current orders for a ship contain HoldFire? If not, then you should create a new map to hold the hold fire state of the player's ships. (AI.h has plenty of maps defined for holding ship specific AI information.)

Reply:
honestly this is the first time I touch orders but it seems like they get overwritten? so yeah a map ig